packages/xchain-cosmos/src/client.ts

```diff
@@ -186,14 +180,23 @@ class Client implements CosmosClient, XChainClient {
    * @throws {"Phrase not set"}
    * Throws an error if phrase has not been set before
    * */
-  private getPrivateKey = (): PrivKey => {
-    if (!this.privateKey) {
-      if (!this.phrase) throw new Error('Phrase not set')
+  private getPrivateKey = (index = 0): PrivKey => {
+    if (!this.phrase) throw new Error('Phrase not set')
 
-      this.privateKey = this.sdkClient.getPrivKeyFromMnemonic(this.phrase)
-    }
+    return this.getSDKClient()?.getPrivKeyFromMnemonic(this.phrase, this.getFullDerivationPath(index))
+  }
+  getSDKClient = (): CosmosSDKClient => {
+    return this.sdkClients.get(this.network) || TESTNET_SDK
+  }
 
-    return this.privateKey
+  /**
+   * Get getFullDerivationPath
+   *
+   * @param {number} index the HD wallet index
+   * @returns {string} The bitcoin derivation path based on the network.
+   */
+  getFullDerivationPath(index: number): string {
```

Comment:
Make `index` optional and set `0` as default value.

BTW: `getFullDerivationPath` is a handy helper function. Pls check other clients, they might need such thing too (e.g. `xchain-bitcoinash`)

Reply:
done